bundles/org.openhab.binding.plex/src/main/java/org/openhab/binding/plex/discovery/PlexDiscoveryService.java

```diff
@@ -0,0 +1,55 @@
+/**
+ * Copyright (c) 2010-2022 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.plex.discovery;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.openhab.binding.plex.internal.PlexBindingConstants;
+import org.openhab.binding.plex.internal.handler.PlexServerHandler;
+import org.openhab.core.config.discovery.AbstractDiscoveryService;
+import org.openhab.core.config.discovery.DiscoveryResult;
+import org.openhab.core.config.discovery.DiscoveryResultBuilder;
+import org.openhab.core.thing.ThingTypeUID;
+import org.openhab.core.thing.ThingUID;
+
+/**
+ * @author Brian Homeyer - Initial Configuration
+ */
+public class PlexDiscoveryService extends AbstractDiscoveryService {
+    private final PlexServerHandler bridgeHandler;
+
+    public PlexDiscoveryService(PlexServerHandler bridgeHandler) {
+        super(PlexBindingConstants.SUPPORTED_THING_TYPES_UIDS, 10, false);
+        this.bridgeHandler = bridgeHandler;
```

Comment:
At this point I am failing to see how this would make things easier. But probably due to my lack of knowledge at this point. 

The bridge in this case only shows the things when they are active (for instance playing a movie). This is done by the method `getAvailablePlayers`.

If this should be refactored, could you help me in the right direction?